samples/Test.AppWriter/Program.cs

```diff
@@ -20,99 +27,180 @@ private static IServiceProvider ConfigureServiceProvider()
+        //}
+    //}
+
+    private static void Main(string[] args)
+    {
+        //var = new Option<>(
+        //    name: "",
+        //    description: ""
+        //    );
+        //var interactiveOption = new Option<bool>(
+        //    name: "--interactive",
+        //    description: "Enables interactive mode for MSApp creation"
+        //    );
+        var filePathOption = new Option<FileInfo?>(
+            name: "--filepath",
+            description: "",
+            isDefault: true,
+            parseArgument: result =>
+            {
+                if (result.Tokens.Count == 0)
+                {
+                    result.ErrorMessage = "Must provide a file path.";
+                    return null;
+                    //Console.WriteLine("Using default out path: ");
+                    //return new FileInfo("appname.msapp");
+                }
+#pragma warning disable IDE0007 // Use implicit type
+                string? filePath = result.Tokens.Single().Value;
+#pragma warning restore IDE0007 // Use implicit type
+                if (File.Exists(filePath)) // TODO: Move this validation after input is handled
+                {
+                    Console.WriteLine("Warning: File already exists");
+                    Console.WriteLine("Provided path: " + filePath);
+                    Console.Write("    Overwrite? (y / n): ");
+                    var input = Console.ReadLine();
+                    if (input?.ToLower()[0] == 'y')
+                    {
+                        File.Delete(filePath);
+                        return new FileInfo(filePath);
+                    }
+                    else
+                    {
+                        result.ErrorMessage = "File already exists and overwrite declined, exiting";
+                        return null;
+                    }
+                }
+                else
+                {
+                    return new FileInfo(filePath);
+                }
+            }
+            );
+        var numScreensOption = new Option<int>(
+            name: "--numscreens",
+            description: "",
+            isDefault: true,
+            parseArgument: result =>
+            {
+                if (!result.Tokens.Any())
+                {
+                    result.ErrorMessage = "Must provide a value for number of screens";
+                    return 0; // Ignored.
+                }
+                else if (int.TryParse(result.Tokens.Single().Value, out var number))
+                {
+                    if (number < 0)
+                    {
+                        result.ErrorMessage = "Number of screens must not be a negative number.";
+                    }
+                    return number;
+                }
+                else
+                {
+                    result.ErrorMessage = "Invalid value, unable to convert to integer.";
+                    return 0; // Ignored.
+                }
+            }
+            );
```

Comment:
For simple types like `int`, you should be able to rely on System.CommandLine to do the parsing for you.   

Validation such as the negative number check can be added in a `.AddValidator(...)` call to separate Parsing and Validation logic.

Not that this is a big deal for small example apps, but thought you might want to be aware of the options.